tests/test_api_events_traversal.py

```
@@ -0,0 +1,56 @@
+    secret_dir = tmp_path / "secret"
+    secret_dir.mkdir()
+
+    (secret_dir / "run_state.json").write_text(json.dumps({"status": "running"}))
+    (secret_dir / "events.jsonl").write_text(json.dumps({"event": "secret_event", "data": "TOP_SECRET"}) + "\n")
+
+    # Construct traversal run_id: ../../secret
+    traversal_id = "../../secret"
+
+    # Verify that calling generate_run_events raises ValueError due to path validation
+    # It might fail due to "traversal sequence" check or "invalid characters" check depending on exact input
+    with pytest.raises(ValueError, match="(traversal sequence|invalid characters)"):
```

Ruff (0.14.14)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)